### ISSUE TITLE
Error in shadowserver parser for chargen

### ISSUE THREAD

**Issue by aaronkaplan** [certtools/intelmq#580]
While running the shadowserver parser, came across this error:

```

2016-07-20 18:38:21,886 - openchargen-parser - DEBUG - Received message {'time.observation': '2016-07-20T12:53:47+00:00', 'raw': '...', 'feed.accuracy': 100.0, 'rtir_report_id': 382605, 'feed.name': 'Shadowserver Open Chargen'}.
2016-07-20 18:38:21,886 - openchargen-parser - ERROR - Failed to parse line.
Traceback (most recent call last):
  File "/home/aaron/intelmq/intelmq/lib/bot.py", line 458, in process
    events = list(filter(bool, self.parse_line(line, report)))
  File "/home/aaron/intelmq/intelmq/bots/parsers/shadowserver/parser.py", line 112, in parse_line
    for item in conf.get('required_fields'):
AttributeError: 'NoneType' object has no attribute 'get'
2016-07-20 18:38:21,888 - openchargen-parser - ERROR - Failed to parse line.
Traceback (most recent call last):
  File "/home/aaron/intelmq/intelmq/lib/bot.py", line 458, in process
    events = list(filter(bool, self.parse_line(line, report)))
  File "/home/aaron/intelmq/intelmq/bots/parsers/shadowserver/parser.py", line 112, in parse_line
    for item in conf.get('required_fields'):
AttributeError: 'NoneType' object has no attribute 'get'
2016-07-20 18:38:21,888 - openchargen-parser - ERROR - Failed to parse line.
Traceback (most recent call last):
  File "/home/aaron/intelmq/intelmq/lib/bot.py", line 458, in process
    events = list(filter(bool, self.parse_line(line, report)))
  File "/home/aaron/intelmq/intelmq/bots/parsers/shadowserver/parser.py", line 112, in parse_line
    for item in conf.get('required_fields'):
AttributeError: 'NoneType' object has no attribute 'get'
2016-07-20 18:38:21,889 - openchargen-parser - ERROR - Failed to parse line.
Traceback (most recent call last):
  File "/home/aaron/intelmq/intelmq/lib/bot.py", line 458, in process
    events = list(filter(bool, self.parse_line(line, report)))
  File "/home/aaron/intelmq/intelmq/bots/parsers/shadowserver/parser.py", line 112, in parse_line
    for item in conf.get('required_fields'):
AttributeError: 'NoneType' object has no attribute 'get'
[...]
2016-07-20 18:38:21,948 - openchargen-parser - ERROR - Failed to parse line.
Traceback (most recent call last):
  File "/home/aaron/intelmq/intelmq/lib/bot.py", line 458, in process
    events = list(filter(bool, self.parse_line(line, report)))
  File "/home/aaron/intelmq/intelmq/bots/parsers/shadowserver/parser.py", line 112, in parse_line
    for item in conf.get('required_fields'):
AttributeError: 'NoneType' object has no attribute 'get'
2016-07-20 18:38:21,948 - openchargen-parser - INFO - Dumping message from pipeline to dump file.
2016-07-20 18:38:21,948 - openchargen-parser - ERROR - Bot has found a problem.
Traceback (most recent call last):
  File "/home/aaron/intelmq/intelmq/lib/bot.py", line 125, in start
    self.process()
  File "/home/aaron/intelmq/intelmq/lib/bot.py", line 466, in process
    self._dump_message(exc, self.recover_line(line))
  File "/home/aaron/intelmq/intelmq/lib/bot.py", line 324, in _dump_message
    new_dump_data[timestamp]["message"] = message.serialize()
AttributeError: 'str' object has no attribute 'serialize'
```

EDIT by @sebix: Trimmed error messages radically and removed sensitive data.

**Comment by sebix** [comment]
Incomplete configuration. Set `feedname` for the parser to `"Open-Chargen"`. See docs: https://github.com/certtools/intelmq/tree/master/intelmq/bots/parsers/shadowserver#how-to-use-the-parser

**Comment by aaronkaplan** [comment]
> On 20 Jul 2016, at 19:02, Sebastian <notifications@github.com> wrote:
> 
> Incomplete configuration. Set feedname for the parser to "Open-Chargen". See docs: https://github.com/certtools/intelmq/tree/master/intelmq/bots/parsers/shadowserver#how-to-use-the-parser
> 

Nope. tested that of course way before.
That's not the reason.

feedname was set correctly.
I was simply documenting this so that I can take a look at it tomorrow.

**Comment by dmth** [comment]
If the `feedname` is not set correctly the parser should stop in `init` (l. 35). Is there a way to start the bot / rerun the bot and skip the `init` function and run `parse_line` directly? In this case the assignment `conf = self.sparser_config` (l. 83) would assign `None` to `conf` which will lead to the reported error in line 112.

 I'm wondering, the error should occur in line 111 when using the recent upstream code and not in line 112...

**Comment by dmth** [comment]
when using the most recent version from certtools/master of this parser ic cannot reproduce this issue... but I receive a `ValueError: invalid literal for int() with base 10: ''` which is caused by the chosen conversion function `int` in the chargen config.

**Comment by dmth** [comment]
see https://github.com/certtools/intelmq/pull/581 use `convert_int` instead of `int()`

**Comment by aaronkaplan** [comment]
fixed in 233b7728b841af040e9c07a12f6ad5dc4aa679f6 . Closing this issue.
